usaspending_api/references/v2/views/city.py

```diff
@@ -0,0 +1,113 @@
+from django.conf import settings
+from rest_framework.response import Response
+from collections import OrderedDict
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.views import APIDocumentationView
+
+from usaspending_api.common.elasticsearch.client import es_client_query
+from usaspending_api.search.v2.elasticsearch_helper import es_sanitize
+from usaspending_api.common.validator.tinyshield import validate_post_request
+
+
+models = [
+    {
+        "name": "filter|country_code",
+        "key": "filter|country_code",
+        "type": "text",
+        "text_type": "search",
+        "optional": False,
+    },
+    {
+        "key": "filter|state_code",
+        "name": "fitler|state_code",
+        "type": "text",
+        "text_type": "search",
+        "optional": True,
+        "default": None,
+        "allow_nulls": True,
+    },
+    {
+        "key": "filter|scope",
+        "name": "filter|scope",
+        "type": "enum",
+        "enum_values": ("recipient_location", "primary_place_of_performance"),
+        "optional": False,
+    },
+    {"key": "search_text", "name": "search_text", "type": "text", "text_type": "search", "optional": False},
+    {"key": "limit", "name": "limit", "type": "integer", "max": 500, "optional": True, "default": 10},
+]
+
+
+@validate_post_request(models)
+class CityAutocompleteViewSet(APIDocumentationView):
+    """
+    endpoint_doc: usaspending_api/api_contracts/autocomplete/City.md
+    """
+
+    @cache_response()
+    def post(self, request, format=None):
+
+        search_text = es_sanitize(request.data["search_text"])
+        country = es_sanitize(request.data["filter"]["country_code"])
+        state = es_sanitize(request.data["filter"]["state_code"])
+        scope = "recipient_location" if request.data["filter"]["scope"] == "recipient_location" else "pop"
+        limit = request.data["limit"]
+        return_fields = ["{}_city_name".format(scope), "{}_state_code".format(scope)]
+        query_string = create_es_search("wildcard", scope, search_text, country, state)
+
+        query = {
+            "_source": return_fields,
+            "size": 0,
+            "query": query_string,
+            "aggs": {
+                "cities": {
+                    "terms": {"field": "{}.keyword".format(return_fields[0]),
+                              "size": 50000,
+                              "order": {
+                                 "max_score": "desc"
+                         }},
+                    "aggs": {
+                        "max_score": {"max": {"script": "_score"}},
+                        "states": {"terms": {"field": return_fields[1], "size": 50000}}
+
+                        },
+                }
+            }
+        }
+
+        hits = es_client_query(index="{}*".format(settings.TRANSACTIONS_INDEX_ROOT), body=query, retries=10)
```

Comment:
@IsaacRay - what is the name of the index here, and the alias to be used for it? In this case, shouldn't the alias cover wildcard-names of indexes it hits? ... such that we should be using the alias here and not need wildcards at the end?

Reply:
@kwhickey The index names does not matter for the API view logic.
The ES alias prefix is used by the API view logic and is set in `settings.TRANSACTIONS_INDEX_ROOT` with a default value of `future-transactions`. The wildcard is necessary since there are several aliases for the different award types